src/main/java/com/microsoft/azure/servicebus/jms/AadAuthentication.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.servicebus.jms;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+
+public class AadAuthentication {
```

Comment:
>AadAuthentication [](http://example.com/codeflow?start=13&length=17)

Can this implement the TokenCredential class instead?

Reply:
we will just be repeating the logic.